js/wazirx.js

```diff
@@ -0,0 +1,709 @@
+        await this.loadMarkets ();
+        const request = {};
+        let market = undefined;
+        if (symbol !== undefined) {
+            market = this.market (symbol);
+            request['symbol'] = market['id'];
+        }
+        const response = await this.spotV1PrivateGetOpenOrders (this.extend (request, params));
+        // [
+        //     {
+        //         "id": 28,
+        //         "symbol": "wrxinr",
+        //         "price": "9293.0",
+        //         "origQty": "10.0",
+        //         "executedQty": "8.2",
+        //         "status": "cancel",
+        //         "type": "limit",
+        //         "side": "sell",
+        //         "createdTime": 1499827319559,
+        //         "updatedTime": 1499827319559
+        //     },
+        //     {
+        //         "id": 30,
+        //         "symbol": "wrxinr",
+        //         "price": "9293.0",
+        //         "stopPrice": "9200.0",
+        //         "origQty": "10.0",
+        //         "executedQty": "0.0",
+        //         "status": "cancel",
+        //         "type": "stop_limit",
+        //         "side": "sell",
+        //         "createdTime": 1499827319559,
+        //         "updatedTime": 1507725176595
+        //     }
+        // ]
+        const orders = this.parseOrders (response, market, since, limit);
+        return orders;
+    }
+
+    async cancelAllOrders (symbol = undefined, params = {}) {
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' cancelAllOrders requires a `symbol` argument');
+        }
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'symbol': market['id'],
+        };
+        return await this.spotV1PrivateDeleteOpenOrders (this.extend (request, params));
+    }
+
+    async cancelOrder (id, symbol = undefined, params = {}) {
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' cancelOrder requires a `symbol` argument');
+        }
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'symbol': market['id'],
+            'orderId': id,
+        };
+        const response = await this.spotV1PrivateDeleteOrder (this.extend (request, params));
+        return this.parseOrder (response);
+    }
+
+    async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'symbol': market['id'],
+            'side': side,
+            'type': type,
+            'quantity': amount,
+        };
+        if ((type === 'limit') || (type === 'stop_limit')) {
+            if (price === undefined) {
+                throw new ArgumentsRequired (this.id + ' createOrder requires a price argument for ' + type + ' orders');
```

Comment:
Let's remove this check for `price === undefined`.

Reply:
I'll but why? (since the order will fail without it) @kroitor

Reply:
We can actually move it higher up if they accept limit-price orders only.

Reply:
let's take into consideration that the only order types supported require the price

Reply:
Yes, I mean, wrapping it inside another if doesn't make much sense:

```JavaScript
if ((type === 'limit') || (type === 'stop_limit')) {
    if (price === undefined) {
        // ...
    }
}
```

Let's transform that to the following:

```JavaScript
if ((type === 'limit') || (type === 'stop_limit')) {
    throw new ExchangeError (this.id + ' createOrder() supports limit and stop_limit orders only');
}
if (price === undefined) {
    throw new ExchangeError (this.id + ' createOrder() requires a price argument');
}
```

Also, note that the actual type depends on whether the `stopPrice` param is defined below – we should be flexible with it.